### PR TITLE
Rename "VisualLinker" to "PdfVisualParser" to welcome "HocrVisualParser"

### PR DESCRIPTION
**Pull request by HiromuHota**
## Description of the problems or issues

**Is your pull request related to a problem? Please describe.**

As #509 now became a big PR, I'll break it down into smaller PRs.
To support hOCR (#476), this PR renames `VisualLinker` to `PdfVisualParser` to welcome `HocrVisualParser`, which will be submitted in a separate PR.

**Does your pull request fix any issue.**

N/A.

## Description of the proposed changes

- Renamed `VisualLinker` to `PdfVisualParser`, which takes `pdf_path` as an argument
    - `pdf_path` now should be a path to a directory containing PDF files.
    - The PDF file's name should be the same as *document_name* + ".pdf" (or ".PDF")
- Changed `Parser`'s signature:
    - Renamed `vizlink` to `visual_parser`.
    - Removed `pdf_path`. Now this is required only by `PdfVisualParser`.
    - Removed `visual`. Use `visual_parser` if visual information is to be parsed.

## Test plan
A clear and concise description of how you test the new changes.

## Checklist

* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] I have updated the CHANGELOG.rst accordingly.

### CODE REVIEW

**Review by senwu — COMMENTED** [review]
Left two comments. Otherwise looks good to me.

File: tests/parser/test_parser.py [hunk 1]
```diff
@@ -797,7 +814,7 @@ def test_parser_skips_and_flattens():
 def test_parser_no_image():
     """Unit test of Parser on a single document that has a figure without image."""
     docs_path = "tests/data/html_simple/no_image.html"
-    pdf_path = "tests/data/pdf_simple/no_image.pdf"
+    pdf_path = "tests/data/pdf_simple/"
```
[the remark this answers is not in the document]
Comment (HiromuHota) [reply tests/parser/test_parser.py:817]: Yes, that's the assumption. I'll highlight this change in the changelog. Thanks.

File: tests/parser/test_parser.py [hunk 2]
```diff
@@ -843,7 +859,7 @@ def parse(docs_path, pdf_path):
 
     # Test that doc_path is a file path, and pdf_path is a file path
```
[the remark this answers is not in the document]
Comment (HiromuHota) [reply tests/parser/test_parser.py]: Good catch! This has been (nicely) resolved by 78a8f5088500afa94aec00ce2c99f574dba88b77.

File: tests/parser/test_parser.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (HiromuHota) [reply tests/parser/test_parser.py:838]: Resolved by CHANGELOG.rst.

**Comment by HiromuHota** [comment]
@SenWu thanks for your review and comments. Please review again and merge it when it's ready.

**Review by senwu — APPROVED** [review]

**Pull request merged**